shell/demo/take1.sh

```diff
@@ -0,0 +1,137 @@
+    test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)
+    if [ -z "$test" ];then
+        echo -e "${RED}[!] Failed to download test${NC}"
+        exit 1
+    else
+        echo -e "${GREEN}[✓] Wrote to temporary file: ${_temp}${NC}"
+        chmod +x $_temp
+    fi
+}
+
+function execute_test {
+    local _temp=$1
+    $_temp
+    local _res=$?
+    if ( echo "100 9 17 18 105 127" | grep -w -q $_res );then
+        echo -e "\n${GREEN}[✓] Result: control test passed${NC}"
+    elif [ $_res -eq 101 ];then
+        echo -e "\n${RED}[!] Result: control test failed${NC}"
+    else
+        echo -e "\n${RED}[!] An unexpected error occurred${NC}"
+    fi
+    return $_res
+}
+
+function execute_cleanup {
+    local _temp=$1
+    local _temp_dir=$(dirname $_temp)
+    rm -f $_temp ${_temp_dir}/09a79e5e20fa4f5aae610c8ce3fe954029a91972b56c6576035ff7e0ec4c1d14.elf ${_temp_dir}/malicious.xlsm ${_temp_dir}/colour-blind.py
+    local _res=$?
+    if [ $_res -eq 0 ];then
+        echo -e "${GREEN}[✓] Clean up is complete${NC}"
+    else
+        echo -e "${RED}[!] Clean up failed${NC}"
+    fi
+}
+
+function post_results {
+    local _dat="$1:$2"
+    curl -sfSL -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${_dat}" $PRELUDE_API
+}
+
+function run_demo {
+    local _i=$1
+    local _test_id=${tests[$_i]}
+    local _temp=$(mktemp)
+    echo -e "###########################################################################################################\n"
+    get_info $_test_id "${names[$_i]}"
+    echo -e "\n###########################################################################################################\n"
+#    sleep 2
+    echo -e "Starting test at: $(date +"%T")"
+    echo -e "\n-----------------------------------------------------------------------------------------------------------"
+#    sleep 2
+    echo -e "[0] Conducting relevance test\n"
+    check_relevance
+    echo -e "-----------------------------------------------------------------------------------------------------------"
+    echo -e "[1] Downloading test\n"
+    download_test $_test_id $_temp
+    echo -e "-----------------------------------------------------------------------------------------------------------"
+    echo -e "[2] Executing test\n"
+    execute_test $_temp
+    local _res=$?
+    echo -e "-----------------------------------------------------------------------------------------------------------"
+    echo -e "[3] Running cleanup\n"
+    execute_cleanup $_temp
+    post_results $_test_id $_res
+    if ( echo "100 9 17 18 105 127" | grep -w -q $_res );then
+        results+=( "${GREEN}${names[$i]}\tPROTECTED${NC}" )
+    elif [ $_res -eq 101 ];then
+        results+=( "${RED}${names[$i]}\tUNPROTECTED${NC}" )
+    else
+        results+=( "${YELLOW}${names[$i]}\tERROR${NC}" )
```

Comment:
colors aren't working for me.. know why?

Reply:
really? that's a shame. The internet told me `tput` is more universal/ better than escape sequences.. but if it's already not working on yours I can move back